src/publisher/management/commands/ingest.py

```diff
@@ -33,28 +34,51 @@ def clean_up(print_queue):
     print_queue.put('STOP')
     reset_queries()
 
-def error(print_queue, errtype, message, log_context):
-    struct = {
-        'status': errtype,
-        'message': message
-    }
+def error(print_queue, errtype, code, message, force, dry_run, log_context, **moar):
+    struct = OrderedDict([
+        ('status', errtype), # final status of request (ingested, published, validated, invalid, error)
+
+        ('code', code), # the error classification (bad request, unknown, parse error, etc)
+        ('comment', codes.explain(code)), # a generic explanation about the error code
+
+        ('message', message), # an explanation of the actual error
+
+        ('trace', None), # optional tracing information. if validation error, it should be error and it's context
+
+        ('dry-run', dry_run),
+        ('force', force),
+
+    ])
+    struct.update(moar)
     log_context['status'] = errtype
     LOG.error(message, extra=log_context)
     write(print_queue, struct)
     clean_up(print_queue)
     sys.exit(1)
 
-def success(print_queue, action, av, log_context, dry_run=False):
-    status = INGESTED if action == INGEST else PUBLISHED
-    attr = 'datetime_published' if status == PUBLISHED else 'datetime_record_updated'
-    struct = {
-        'status': status,
-        'id': None if dry_run else av.article.manuscript_id,
-        'datetime': getattr(av, attr),
-        'message': "(dry-run)" if dry_run else None,
+def error_from_err(print_queue, errtype, errobj, force, dry_run, log_context):
+    return error(print_queue, errtype, errobj.code, errobj.message, force, dry_run, log_context, trace=errobj.trace)
+
+def success(print_queue, action, av, force, dry_run, log_context):
+    lu = {
+        INGEST: INGESTED,
+        PUBLISH: PUBLISHED,
+        INGEST_PUBLISH: PUBLISHED,
     }
+    status = orig_status = lu[action]
+    if dry_run:
+        # this is a dry run, we didn't actually ingest or publish anything
+        status = VALIDATED
+
+    attr = 'datetime_published' if orig_status == PUBLISHED else 'datetime_record_updated'
+    struct = OrderedDict([
+        ('status', status),
+        ('id', log_context['msid']), # good idea? #'id': None if dry_run else av.article.manuscript_id,
+        ('datetime', getattr(av, attr)),
```

Comment:
that value is already a little murky. if the action is PUBLISH, it's the datetime of the publication event, the one that is stored in lax, which is controlled by lax if it's a non v1, but controlled by the xml/production if it's a v1. on ingest it's the time it was created, on validate it's when the job is complete - so, different meanings for different actions but ultimately just a datetime stamp.